### PR TITLE
access-control: implement in-memory rate limiting

### PR DESCRIPTION
**Pull request by gioelecerati**
This should fix rate limiting, handling it in memory in catalyst instead of Studio
Goes along with https://github.com/livepeer/studio/pull/1982

### CODE REVIEW

**Review by thomshutt — APPROVED** [review]
Got it, thanks! LGTM once linting is fixed

File: handlers/accesscontrol/access-control.go
Comment (pwilczynskiclearcode) [inline handlers/accesscontrol/access-control.go:128]: You wont' unlock `mux` if returning in here? right?
It's good to use `defer c.mux.Unlock()` like you did in cleanupCache() if not optimising for earliest-possible unlock

Comment (gioelecerati) [reply handlers/accesscontrol/access-control.go:129]: Right, good point! I added the defer unlock and fixed the cucumber tests failing - will add a unit test in a follow up PR

File: handlers/accesscontrol/access-control.go
Comment (thomshutt) [inline handlers/accesscontrol/access-control.go:133]: Do we need this double unlock?

File: handlers/accesscontrol/access-control.go
Comment (thomshutt) [inline handlers/accesscontrol/access-control.go:254]: Could you add a comment on what these return params are please?

File: handlers/accesscontrol/access-control.go
Comment (thomshutt) [inline handlers/accesscontrol/access-control.go:211]: Shall we put something in to only let this run every X seconds? Otherwise we're running over this locking loop really frequently in a high traffic situation

File: handlers/accesscontrol/access-control.go
Comment (gioelecerati) [reply handlers/accesscontrol/access-control.go:254]: Sure, adding them

Considering that they grew and soon they are going to include other params (refreshInterval, for example) I want to put them into a separate `type` in the next PR

File: handlers/accesscontrol/access-control.go
Comment (gioelecerati) [reply handlers/accesscontrol/access-control.go:133]: Right! removed it on line 133

**Pull request merged**
